packages/core/data-transfer/src/strapi/providers/local-source/assets.ts

```diff
@@ -1,36 +1,47 @@
 import { join } from 'path';
-import https from 'https';
-import http from 'http';
 import { Duplex, PassThrough, Readable } from 'stream';
+import * as webStream from 'stream/web';
 import { stat, createReadStream, ReadStream } from 'fs-extra';
 import type { LoadedStrapi } from '@strapi/types';
 
 import type { IAsset } from '../../../../types';
 
-const protocolForPath = (filepath: string) => {
-  return filepath?.startsWith('https') ? https : http;
-};
-
 function getFileStream(filepath: string, isLocal = false): PassThrough | ReadStream {
   if (isLocal) {
     return createReadStream(filepath);
   }
 
   const readableStream = new PassThrough();
 
-  protocolForPath(filepath)
-    .get(filepath, (res) => {
-      if (res.statusCode !== 200) {
-        readableStream.emit(
-          'error',
-          new Error(`Request failed with status code ${res.statusCode}`)
-        );
+  // fetch the image from remote url and stream it
+  fetch(filepath)
```

Comment:
```suggestion
  strapi.fetch(filepath)
```
This should be going through strapi.fetch, but not sure if we have access here. Will update later.